### PR TITLE
implemented task and tests

### PR DESCRIPTION
**Pull request by ivadimko**

### CODE REVIEW

File: src/arrayMethodSort.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+const applyCustomSort = require('./arrayMethodSort');
+
+applyCustomSort();
+
+test('sort2 is added to [].__proto__', () => {
+  expect([].sort2)
+    .toBeInstanceOf(Function);
+});
+
+test('sort return a link to the original array', () => {
```
Comment (roman-mate) [inline src/arrayMethodSort.test.js]: suggestion: sort returns original array.

File: src/arrayMethodSort.test.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  const result = source.sort2();
+  expect(source)
+    .toBe(result);
+});
+
+test('the original array is sorted', () => {
+  const source = [3, 12, 2, 11];
+  source.sort2();
+  expect(source)
+    .toEqual([11, 12, 2, 3]);
```
Comment (roman-mate) [inline src/arrayMethodSort.test.js]: this is sorted as string. WAI?

File: src/arrayMethodSort.test.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+test('sort return a link to the original array', () => {
+  const source = [3, 12, 2, 11];
+  const result = source.sort2();
+  expect(source)
+    .toBe(result);
+});
+
+test('the original array is sorted', () => {
+  const source = [3, 12, 2, 11];
```
Comment (roman-mate) [inline src/arrayMethodSort.test.js]: this array is not really sorted.

File: src/arrayMethodSort.test.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  source.sort2();
+  expect(source)
+    .toEqual([11, 12, 2, 3]);
+});
+
+test('sort numbers: [3, 12, 2, 11].sort2()', () => {
+  expect([3, 12, 2, 11].sort2())
+    .toEqual([11, 12, 2, 3]);
```
Comment (roman-mate) [inline src/arrayMethodSort.test.js]: 12 is more than 11.

Comment (roman-mate) [reply src/arrayMethodSort.test.js]: maybe update test description to smth like: "numbers are sorted as strings by default"

File: src/arrayMethodSort.test.js (hunk 3 above)
Comment (ivadimko) [reply src/arrayMethodSort.test.js]: updated description to 'the array is sorted'

**Pull request merged**